### PR TITLE
V0.12.0 rc

### PR DESCRIPTION
**Pull request by rhornung67**
# Summary

- This PR merges develop plus changes in v0.12.0-rc branch into main so we can tag main for the release.
- Note that there are many changes. The differences between develop and v0.12.0-rc are much smaller and this is what should be reviewed as all other content in develop has been approved in numerous PRs. However, please approve this PR based on those changes. Here is the link to the PR with the changes between develop and v0.12.0-rc: https://github.com/LLNL/RAJA/pull/876
- After this is merged into main, the release tag will be created and a PR will be put up to merge back to develop.

### CODE REVIEW

**Comment by gzagaris** [comment]
At last, the wait is (almost) over! 🎉

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -10,6 +10,108 @@
 Version vxx.yy.zz -- Release date 20yy-mm-dd
 ============================================
```
Comment (rchen20) [inline RELEASE_NOTES.md:11]: Take this out?

```suggestion
```

File: RELEASE_NOTES.md [hunk 2]
```diff
@@ -10,6 +10,108 @@
...
+      * Added support for dynamic plug-ins in RAJA, which enable the use of
+        things like Kokkos Performance Profiline Tools to be used with RAJA
+        (https://github.com/kokkos/kokkos-tools)
+      * Added ability to pass a resource object to RAJA::forall methods to
+        enable asynchronous execution for CUDA and HIP back-ends.
+      * Added "Multi-view" that works like a regular view, except that it
+        can wrap multiple arrays so their accesses can share index arithmetic.
+      * Introduced RAJA "Teams" concept as an experimental feature. This
+        enables hierarchical parallelism and additional nested loop patterns
+        beyond what RAjA::kernel supports. Please note that this is very much
```
Comment (mdavis36) [inline RELEASE_NOTES.md:37]: RAjA -> RAJA

File: RELEASE_NOTES.md (hunk 1 above)
Comment (rhornung67) [reply RELEASE_NOTES.md:11]: That's always been in the file. It's a placeholder for starting the next release.

File: RELEASE_NOTES.md (hunk 2 above)
Comment (rhornung67) [reply RELEASE_NOTES.md:37]: Thanks @mdavis36. Will fix.

**Comment by rhornung67** [comment]
@mdavis36 @rchen20 @davidbeckingsale @MrBurmark I addressed all the typos you reported and updated some build scripts that now work with CUDA 11. Please review if you want to take another look before this is merged. Thank you.

File: RELEASE_NOTES.md [hunk 3]
```diff
@@ -10,6 +10,108 @@
...
+
+This release contains new features, notable changes, and bug fixes. Please
+see the RAJA user guide for more information about items in this release.
+
+Notable changes include:
+
+  * Notable repository change:
+      * The 'master' branch in the RAJA git repo has been renamed to 'main'.
+
+  * New features:
```
Comment (rchen20) [inline RELEASE_NOTES.md:24]: @rhornung67 @MrBurmark Did we want to make users aware of "sort" as a New Feature? Maybe a change like the following (just the first sentence from the sort documentation):
```suggestion
      * New RAJA "sort" capability added. This provides portable parallel sort operations, which are basic parallel algorithm building blocks.
```

Comment (rhornung67) [reply RELEASE_NOTES.md:24]: I thought I had it in there. Sorry. It is now added.

**Pull request merged**